### PR TITLE
allow calendar service to be called from other services

### PR DESCRIPTION
**Pull request by sdetweil**
make routines use external data

####  Fixes Related issues
- add related
- issues here

####  Checklist
- [ x] I have read and understand the CONTRIBUTIONS.md file
- [ x] I have searched for and linked related issues
- [ x] I have added config.schema.json file if config option are required.
- [x ] I am NOT targeting master branch

### CODE REVIEW

File: plugins/calendar/service.js [hunk 1]
```diff
@@ -148,27 +151,17 @@
 							recuring_event.startName = startDate.calendar().toUpperCase();
 							recuring_event.end = endDate;
 							recuring_event.endName = endDate.subtract(1, 'seconds').calendar().toUpperCase();
-
-							if (recuring_event.startName && recuring_event.endName) {
-								if (recuring_event.startName == recuring_event.endName) {
-									recuring_event.label = recuring_event.startName
```
Comment (evancohen) [inline plugins/calendar/service.js]: Why did you remove this check?

**Comment by sdetweil** [comment]
in the original implementation the data is stored in a STATIC array created INSIDE the service method. 
so, if YOU call it and I call it at the same time, we stomp on each others results. 

here is how I call it

var events = [];
// get the events from the default calendars
CalendarService.getCalendarEvents(null, events).then(

here is how the original calls it
 CalendarService.getCalendarEvents().then(function () {
the 1st thing the original code did was clear the results
 service.events = [];

if I call it while your invocation is in progress, then I destroy your results..

File: plugins/calendar/service.js (hunk 1 above)
Comment (sdetweil) [inline plugins/calendar/service.js]: hm.. I did a diff of my earlier pre-plugin implementation, and found no changes there..  
so that should not have been removed. 

all this change does is remove the hard coded data storage name (service.events) from all the routines and have them take a passed in storage area for the array of events.

**Comment by kurtdb** [comment]
An angular service is by default a singleton (only 1 instance will exist at all times), so if 2 controllers call the same function at the same time 1 will have to wait.

Your concern with mixing results is actually a non-issue.

**Comment by sdetweil** [comment]
interesting, but my code calling the calendar service is not via a controller, but from a timer handler..
I got mixed results when I started doing this, which is why I made these chnages

**Comment by kurtdb** [comment]
A bit late to add this, but the way you should add the time handler is inside the Angular Controller using the $interval variable. That way, you're inside the angular lifecycle and everything is handled for you.

**Comment by sdetweil** [comment]
that is how it (the timer tick) is done..  using the interval timer

**Comment by justbill2020** [comment]
this is a single user application....

if you're handling this as a interval timer in the plugin controller this PR isn't necessary.

**Comment by sdetweil** [comment]
single user and multi-threaded are different topics..  

there are two parts of this PR.. 

1. change the buffer
2. use parameters for the days/count of calendar entries..

we are talking about item 1

I still need item 2.

**Comment by sdetweil** [comment]
this PR is closed.. all in the other #602
